scripts/west_commands/runners/stm32cubeprogrammer.py

```diff
@@ -0,0 +1,199 @@
+            self._extra_opts += opts
+
+        # add required library loader path to the environment (Linux only)
+        if platform.system() == "Linux":
+            os.environ["LD_LIBRARY_PATH"] = str(self._cli.parent / ".." / "lib")
+
+    @staticmethod
+    def _get_stm32cubeprogrammer_path() -> Path:
+        """Obtain path of the STM32CubeProgrammer CLI tool."""
+
+        if platform.system() == "Linux":
+            return (
+                Path.home()
+                / "STMicroelectronics"
+                / "STM32Cube"
+                / "STM32CubeProgrammer"
+                / "bin"
+                / "STM32_Programmer_CLI"
+            )
+
+        if platform.system() == "Windows":
+            return (
+                Path(os.environ["PROGRAMFILES"])
+                / "STMicroelectronics"
+                / "STM32Cube"
+                / "STM32CubeProgrammer"
+                / "bin"
+                / "STM32_Programmer_CLI.exe"
+            )
+
+        if platform.system() == "Darwin":
+            return (
+                Path("/Applications")
+                / "STMicroelectronics"
+                / "STM32Cube"
+                / "STM32CubeProgrammer"
+                / "STM32CubeProgrammer.app"
+                / "Contents"
+                / "MacOs"
+                / "bin"
+                / "STM32_Programmer_CLI"
+            )
+
+        raise NotImplementedError("Could not determine STM32_Programmer_CLI path")
+
+    @classmethod
+    def name(cls):
+        return "stm32cubeprogrammer"
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={"flash"}, erase=True)
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument(
+            "--port",
+            type=str,
+            required=True,
+            help="Interface identifier, e.g. swd, jtag, /dev/ttyS0...",
+        )
+        parser.add_argument(
+            "--frequency", type=int, required=False, help="Programmer frequency in KHz"
+        )
+        parser.add_argument(
+            "--reset-mode",
+            type=str,
+            required=False,
+            choices=["sw", "hw", "core"],
+            help="Reset mode",
+        )
+        parser.add_argument(
+            "--extra-connect-opts",
+            type=str,
+            required=False,
+            help="Additional options for the --connect argument",
+        )
+        parser.add_argument(
+            "--cli",
+            type=Path,
+            required=False,
+            help="STM32CubeProgrammer CLI tool path",
+        )
+        parser.add_argument(
+            "--use-elf",
+            action="store_true",
+            required=False,
+            help="Use ELF file when flashing instead of HEX file",
+        )
+        parser.add_argument(
+            "--extra-opts",
+            default=[],
+            action="append",
+            help="Additional options for STM32_Programmer_CLI",
```

Comment:
Similarly it feels like this could be renamed --tool-opt for consistency with the other runners.

Reply:
fixed